src/blog/2024/01/taking-it-further-with-node-red.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Getting Started with Node-RED
+subtitle: Node-RED is one of the easiest ways to program ever created but everyone needs a little help
+description: In this article we are going to explain how you can store data outside of msg.payload for later use
+date: 2024-02-12
+image: /blog/2023/01/images/getting-started-nr.png
+authors: ["rob-marcer"]
+tags:
+    - posts
+    - node-red
+    - how-to
+---
+
+It's quite straightforward to pass plenty of useful data with each message (msg) in your flows. Not only can you store information in msg.payload, but you can also place information in any other named object, for instance, msg.store.
+
+<!--more-->
+
+In this article, we will explore some of the better solutions for storing and retrieving transactional information in your flows.
+
+#### Storing data outside of msg.payload
+
+In this example, we have data in msg.payload as well as in msg.later.
+
+![Storing data outside of msg.payload](./images/data-outside-msg-payload.gif "Storing data outside of msg.payload")
+
+If you want your debug to display the full content of the message, change the output to 'complete message object' as shown above.
+
+You can import the flow using this code.
+
+```json
+[{"id":"aaa1e17e5f158004","type":"inject","z":"67746003c844dbc4","name":"Inject the message","props":[{"p":"payload"},{"p":"later","v":"A string I want to be able to use later in my flow","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"Hello World","payloadType":"str","x":690,"y":140,"wires":[["bce1bf09736125b7"]]},{"id":"bce1bf09736125b7","type":"debug","z":"67746003c844dbc4","name":"debug 1","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":880,"y":140,"wires":[]}]
+```
+
+Storing data outside of msg.payload can be very useful when you need access to that data later in your flow. You may notice that many nodes overwrite the content of msg.payload, so putting your data elsewhere is essential otherwise it will be overwritten and lost.
```

Comment:
```suggestion
Storing data outside of msg.payload can be very useful when you need access to that data later in your flow. You may notice that many nodes overwrite the content of msg.payload, so putting your data elsewhere is essential otherwise, it will be overwritten and lost.
```